storage/innobase/srv/srv0srv.cc

```diff
@@ -1427,6 +1428,13 @@ srv_export_innodb_status(void)
 		btr_scrub_total_stat(&scrub_stat);
 	}
 
+#ifdef BTR_CUR_HASH_ADAPT
+	mem_adaptive_hash = 0;
+#endif
+	mem_dictionary = (dict_sys ? ((dict_sys->table_hash->n_cells + 
+		dict_sys->table_id_hash->n_cells + dict_sys->temp_id_hash->n_cells) * 
+		sizeof(hash_cell_t) + dict_sys_get_size()) : 0);
+
```

Comment:
MDEV-19513 in MariaDB 10.4.5 removed the pointer indirection for `dict_sys`. Basically, `dict_sys->` must be replaced with `dict_sys.` and `dict_operation_lock` with `dict_sys.latch`.

That change is now also merged to the 10.5 branch. 893472d005b6c62209b15b0955af01a55069d8ef in 10.5 removed the indirection of `export_vars.innodb_available_undo_logs`, like this:
```diff
diff --git a/storage/innobase/handler/ha_innodb.cc b/storage/innobase/handler/ha
_innodb.cc
index 5a79782f5b3..0ccd4384ab0 100644
--- a/storage/innobase/handler/ha_innodb.cc
+++ b/storage/innobase/handler/ha_innodb.cc
@@ -1054,8 +1054,7 @@ static SHOW_VAR innodb_status_variables[]= {
   (char*) &export_vars.innodb_num_open_files,            SHOW_LONG},
   {"truncated_status_writes",
   (char*) &export_vars.innodb_truncated_status_writes,   SHOW_LONG},
-  {"available_undo_logs",
-  (char*) &export_vars.innodb_available_undo_logs,        SHOW_LONG},
+  {"available_undo_logs", &srv_available_undo_logs, SHOW_ULONG},
   {"undo_truncations",
   (char*) &export_vars.innodb_undo_truncations,           SHOW_LONG},
 
```
I think that it would be good to do something similar for other variables, where applicable.

Reply:
Sure, will do.
Sorry for the delays, I've been sidetracked by other commitments this past week. Hope to focus on this next week.